src/Features/CSharp/Portable/Copilot/CSharpCopilotNotImplementedMethodFixProvider.CodeGenerator.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.CSharp.Copilot;
+
+internal sealed partial class CSharpCopilotNotImplementedMethodFixProvider
+{
+    internal static class CodeGenerator
+    {
+        internal static void GenerateCode(SyntaxEditor editor, SyntaxNode throwNode, string codeBlockSuggestion)
+        {
+            if (string.IsNullOrWhiteSpace(codeBlockSuggestion))
+                return;
+
+            try
+            {
+                // Find the containing method declaration
+                var methodDeclaration = throwNode.Parent.FirstAncestorOrSelf<MethodDeclarationSyntax>();
```

Comment:
1. how do you know you have a containing method decl
2. we shlud not lmiit things to just methods.